### ISSUE TITLE
FPS/performance counters

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#2223]
Right now openrct does not have ability to report its performance. Only external tools can show FPS, but this is only when it is running in non-headless mode. Those tools are also usually not portable.

We'd need a set of counters (say one for rendering, one for game logic) which would allow reporting of performance, possibly also tracking regressions or improvements.

**Comment by IntelOrca** [comment]
I have added showing of FPS now, but only drawing it in-game. It will need to be separated for headless mode. Other performance counters should use a finer performance measuring utility such as ```QueryPerformanceCounter(...)``` and ```QueryPerformanceFrequency(...)``` on Windows.

**Comment by janisozaur** [comment]
Nice! I get anything between 15 and 333 fps on i5-5300U :P

For measuring other things I was thinking about using C++11's `std::chrono`, but haven't investigated feasibilty of it yet. This is what I have used in #2328

**Comment by IntelOrca** [comment]
@janisozaur yes you can use std::chrono, for Windows that uses ```QueryPerformanceCounter(...)``` and ```QueryPerformanceFrequency(...)``` under the hood.

**Comment by Stevoisiak** [comment]
This issue can probably be closed now that there's a built in FPS counter.

**Comment by Nubbie** [comment]
Still valid when we have a FPS counter?

**Comment by janisozaur** [comment]
Yes, to some extent. Reporting performance on headless builds would be nice, but I imagine that can be alleviated by script engine.

I have already added `benchgfx` subcommand, but there should be some more.
